src/buildblock/recon_array_functions.cxx

```diff
@@ -366,35 +372,44 @@ void accumulate_loglikelihood(Viewgram<float>& projection_data,
      accum would be no longer change because of the finite precision.
   */
   double result = 0;
-  const float small_value= 
-    max(projection_data.find_max()*SMALL_NUM, 0.F);
+  const float small_value = max(projection_data.find_max()*SMALL_NUM, 0.F);
   const float max_quotient = 10000.F;
 
+  // Loop over axial positions
   for(int r=rs;r<=re;r++)
   {
     double sub_result=0; // use this for total result for this r, reducing numerical error
+    // Loop over tangential positions
     for(int b=bs;b<=be;b++)  
-      if(!(
-	   b<bs+rim_truncation_sino ||
-	   b>be-rim_truncation_sino ))
-	{
-          // if (estimated_projections[r][b] == 0)
-          //  std::cerr << "Zero at " << r << ", " << b <<'\n';
-	  const float new_estimate =
-	    max(estimated_projections[r][b], 
-		projection_data[r][b]/max_quotient);
-	  if (projection_data[r][b]<=small_value)
-	    sub_result += - double(new_estimate);
-	  else
-	    sub_result += projection_data[r][b]*log(double(new_estimate)) - double(new_estimate);
-	}
+      if(!( b<bs+rim_truncation_sino || b>be-rim_truncation_sino ))
+      {
+        // if (estimated_projections[r][b] == 0)
+        //  std::cerr << "Zero at " << r << ", " << b <<'\n';
+        const float new_estimate = max(estimated_projections[r][b], projection_data[r][b]/max_quotient);
+        sub_result += compute_data_fit_term(projection_data[r][b], new_estimate, use_KL_divergence, small_value);
```

Comment:
I like putting this in a function, such that we can call it. I'm not convinced about the name though. At some point, we want to have least squares data-fit as well. Any ideas? (`compute_Poisson_data_fit`?)

Reply:
I am bad at naming things and this is a bad name. `compute_Poisson_data_fit` is good. Ill keep thinking.